### PR TITLE
Fixed an issue where fog sky color mode could sample NaNs in the sky cubemap

### PR DESCRIPTION
**Pull request by JulienIgnace-Unity**
### Purpose of this PR
Fixed an issue where fog sky color mode could sample NaNs in the sky cubemap.

Fix this issue: https://fogbugz.unity3d.com/f/cases/1209238/

---
### Testing status

**Manual Tests**: What did you do?
- [x] Opened test project + Run graphic tests locally
- [ ] Built a player
- [ ] Checked new UI names with UX convention
- [ ] Tested UI multi-edition + Undo/Redo + Prefab overrides + Alignment in Preset
- [ ] C# and shader warnings (supress shader cache to see them)
- [ ] Checked new resources path for the reloader (in devloper mode, you have a button at end of resources that check the pathes)

### CODE REVIEW

File: com.unity.render-pipelines.high-definition/Runtime/Sky/SkyManager.cs [hunk 1]
```diff
@@ -797,8 +798,6 @@ public void UpdateEnvironment(HDCamera hdCamera, ScriptableRenderContext renderC
 
             var reflectionTexture = GetReflectionTexture(hdCamera.lightingSky);
             cmd.SetGlobalTexture(HDShaderIDs._SkyTexture, reflectionTexture);
-            float mipCount = Mathf.Clamp(Mathf.Log((float)reflectionTexture.width, 2.0f) + 1, 0.0f, 6.0f);
-            cmd.SetGlobalFloat(HDShaderIDs._SkyTextureMipCount, mipCount);
```
Comment (sebastienlagarde) [inline com.unity.render-pipelines.high-definition/Runtime/Sky/SkyManager.cs]: Please remove string: _SkyTextureMipCount

File: com.unity.render-pipelines.core/ShaderLibrary/ImageBasedLighting.hlsl [hunk 2]
```diff
@@ -8,6 +8,7 @@
 #include "Packages/com.unity.render-pipelines.core/ShaderLibrary/Sampling/Sampling.hlsl"
 
 #ifndef UNITY_SPECCUBE_LOD_STEPS
```
Comment (sebastienlagarde) [inline com.unity.render-pipelines.core/ShaderLibrary/ImageBasedLighting.hlsl]: Do we still need it? UNITY_SPECCUBE_LOD_STEPS

Comment (JulienIgnace-Unity) [reply com.unity.render-pipelines.core/ShaderLibrary/ImageBasedLighting.hlsl]: It's in core and used by URP at least so we can't really remove it.

**Comment by sebastienlagarde** [comment]
@JulienIgnace-Unity is the PR ready? should we retest it?

**Pull request merged**